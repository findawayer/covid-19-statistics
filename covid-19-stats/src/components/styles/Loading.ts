import styled from '@emotion/styled';

// CSS-loaders | Copyright (c) 2014 Luke Haas | MIT license
// https://github.com/lukehaas/css-loaders
export const StyledLoading = styled.div<{ isCentered?: boolean }>`
  position: relative;
  width: 10em;
  height: 10em;
  margin: 55px auto;
  color: #aaa;
  box-shadow: inset 0 0 0 1em;
  transform: translateZ(0);

  ${({ isCentered }) =>
    isCentered &&
    `
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  `}

  &,
  &::before,
  &::after {
    border-radius: 50%;
  }
  &::before,
  &::after {
    position: absolute;
    content: '';
  }
  &::before {
    top: -0.1em;
    left: -0.1em;
    width: 5.2em;
    height: 10.2em;
    background: #fff;
    border-radius: 10.2em 0 0 10.2em;
    transform-origin: 5.1em 5.1em;
    animation: load2 2s infinite ease 1.5s;
  }
  &::after {
    top: -0.1em;
    left: 4.9em;
    width: 5.2em;
    height: 10.2em;
    background: #fff;
    border-radius: 0 10.2em 10.2em 0;
    transform-origin: 0.1em 5.1em;
    animation: load2 2s infinite ease;
  }

  @keyframes load2 {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`;
