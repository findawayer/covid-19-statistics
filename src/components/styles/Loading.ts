import styled from '@emotion/styled';

export const StyledLoading = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;
