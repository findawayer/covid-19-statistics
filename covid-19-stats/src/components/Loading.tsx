import type { FunctionComponent } from 'react';
import React from 'react';

import { StyledLoading } from './styles/Loading';

interface LoadingProps {
  isCentered?: boolean;
}

const Loading: FunctionComponent<LoadingProps> = ({ isCentered }) => (
  <StyledLoading
    isCentered={isCentered}
    role="presentation"
    aria-label="Loading"
  />
);

export default Loading;
