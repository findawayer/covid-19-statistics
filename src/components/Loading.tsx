import type { FunctionComponent } from 'react';
import React from 'react';

import { StyledLoading } from './styles/Loading';

interface LoadingProps {
  size?: number;
}

/* SVG-loaders | 2014 Sam Herbert, MIT License | https://github.com/SamHerbert/SVG-Loaders */
const Loading: FunctionComponent<LoadingProps> = ({ size = 64 }) => {
  return (
    <StyledLoading aria-label="Loading">
      <svg
        width={size}
        height={size}
        viewBox="0 0 38 38"
        xmlns="http://www.w3.org/2000/svg"
        stroke="#3498DB"
      >
        <g fill="none" fill-rule="evenodd">
          <g transform="translate(1 1)" stroke-width="2">
            <circle stroke-opacity=".5" cx="18" cy="18" r="18" />
            <path d="M36 18c0-9.94-8.06-18-18-18">
              <animateTransform
                attributeName="transform"
                type="rotate"
                from="0 18 18"
                to="360 18 18"
                dur="1s"
                repeatCount="indefinite"
              />
            </path>
          </g>
        </g>
      </svg>
    </StyledLoading>
  );
};

export default Loading;
