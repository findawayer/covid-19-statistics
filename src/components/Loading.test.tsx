import { render } from '@testing-library/react';
import Loading from './Loading';

describe('<Loading />', () => {
  it('renders a spinner', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });

  it('renders a spinner with specific size (100)', () => {
    const { container } = render(<Loading size={100} />);
    expect(container).toMatchSnapshot();
  });
});
