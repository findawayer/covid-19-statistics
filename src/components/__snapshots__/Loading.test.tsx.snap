// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders a spinner 1`] = `
<div>
  <div
    aria-label="Loading"
    class="css-1sgbmxk"
  >
    <svg
      height="64"
      stroke="#3498DB"
      viewBox="0 0 38 38"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="2"
          transform="translate(1 1)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            stroke-opacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`<Loading /> renders a spinner with specific size (100) 1`] = `
<div>
  <div
    aria-label="Loading"
    class="css-1sgbmxk"
  >
    <svg
      height="100"
      stroke="#3498DB"
      viewBox="0 0 38 38"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="2"
          transform="translate(1 1)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            stroke-opacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
